Source/OCMock/OCMMacroState.m

```diff
@@ -90,8 +98,7 @@ - (id)initWithRecorder:(OCMRecorder *)aRecorder
 - (void)dealloc
 {
     [recorder release];
-    if(globalState == self)
-        globalState = nil;
+    NSAssert([NSThread currentThread].threadDictionary[OCMGlobalStateKey] != self, @"Unexpected dealloc while set as the global state");
     [super dealloc];
```

Comment:
Yes the object for OCMGlobalStateKey should always be removed from the thread dictionary at this point, the assert was a "just in case" thing.  I don't mind removing it.